extensions/iceberg/src/main/java/io/deephaven/iceberg/layout/IcebergKeyValuePartitionedLayout.java

```diff
@@ -53,32 +53,23 @@ public IcebergKeyValuePartitionedLayout(
 
         // We can assume due to upstream validation that there are no duplicate names (after renaming) that are included
         // in the output definition, so we can ignore duplicates.
-        final MutableInt icebergIndex = new MutableInt(0);
-        final Map<String, Integer> availablePartitioningColumns = partitionSpec.fields().stream()
-                .peek(partitionField -> {
-                    // TODO (deephaven-core#6438): Add support to handle non-identity transforms
-                    if (!partitionField.transform().isIdentity()) {
-                        throw new TableDataException("Partition field " + partitionField.name() + " has a " +
-                                "non-identity transform: " + partitionField.transform() + ", which is not supported");
-                    }
-                })
+        final AtomicInteger icebergIndex = new AtomicInteger(0);
+        // TODO (deephaven-core#6438): Add better support for handling non-identity transforms
+        identityPartitioningColumns = partitionSpec.fields().stream()
```

Comment:
I think we should just do a for-loop instead with int index that is incremented...